src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -0,0 +1,236 @@
+      <v-card-text>
+        <div id="notation-text" class="mb-4 mt-2 default-text">
+          Enter a {{itemName}} that will appear on the ledger for this entity
+        </div>
+        <v-form ref="notationFormRef" v-model="notationFormValid" id="notation-form">
+          <v-textarea
+            v-model="notation"
+            class="text-input-field mb-2"
+            filled
+            :label="itemName"
+            id="notation"
+            rows="4"
+            :rules="notationRules"
+            :counter="notationMaxLength"
+          />
+        </v-form>
+        <div class="default-text">
+          If this filing is pursuant to a court order, enter the court order number. If this filing is pursuant
+          to a plan of arrangement, enter the court order number and select Plan of Arrangement.
+        </div>
+        <CourtOrderPoa
+          id="court-order"
+          @emitCourtNumber="setFileNumber($event)"
+          @emitPoa="setHasPlanOfArrangement($event)"
+          :displaySideLabels="false"
+          :key="courtOrderKey"
+          :autoValidation="enableValidation"
+          :courtOrderNumberRequired="courtOrderNumberRequired"
+          ref="courtOrderPoaRef"
+        />
+      </v-card-text>
+      <v-divider class="mb-4"></v-divider>
+      <v-card-actions class="pt-0">
+        <v-spacer></v-spacer>
+        <div class="form__btns">
+          <v-btn text color="primary"
+            id="dialog-save-button"
+            :loading="saving"
+            @click.native="save()"
+          >Save</v-btn>
+          <v-btn text color="secondary"
+            id="dialog-cancel-button"
+            :disabled="saving"
+            :loading="saving"
+            @click.native="emitClose(false)"
+          >Cancel</v-btn>
+        </div>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins'
+import { mapState } from 'vuex'
+import axios from '@/axios-auth'
+import { CourtOrderPoa } from '@bcrs-shared-components/court-order-poa'
+import { FormIF } from '@/interfaces'
+
+@Component({
+  computed: {
+    ...mapState(['entityIncNo'])
+  },
+  components: {
+    CourtOrderPoa
+  }
+})
+export default class AddStaffNotationDialog extends Mixins(DateMixin) {
+  $refs!: Vue['$refs'] & {
+    courtOrderPoaRef: FormIF,
+    notationFormRef: FormIF
+  }
+
+  /** Prop for the item's name of the dialog. */
+  @Prop() readonly itemName: string
+
+  /** Prop for the item's filing type. */
+  @Prop() readonly filingType: string
+
+  /** Prop to display the dialog. */
+  @Prop() readonly dialog: boolean
+
+  /** Prop to provide attachment selector. */
+  @Prop() readonly attach: string
+
+  /** Prop to require court order number regardless the plan of arrangement. */
+  @Prop() readonly courtOrderNumberRequired: boolean
```

Comment:
Do any of these props need a default value?

Reply:
Yes. It won't be a problem not having a default but I added a default value to be explicit.